tests/zfs-tests/tests/functional/no_space/enospc_df.ksh

```diff
@@ -0,0 +1,62 @@
+#!/bin/ksh -p
+#
+# CDDL HEADER START
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+# CDDL HEADER END
+#
+
+#
+# Copyright (c) 2014, 2016 by Delphix. All rights reserved.
+# Copyright (c) 2019 by Datto Inc. All rights reserved.
+#
+
+. $STF_SUITE/include/libtest.shlib
+. $STF_SUITE/tests/functional/no_space/enospc.cfg
+
+#
+# DESCRIPTION:
+# After filling a filesystem, the df command produces the
+# expected result against the pool root filesystem.
+#
+
+verify_runnable "both"
+
+function cleanup
+{
+	log_must_busy zpool destroy -f $TESTPOOL
+}
+
+log_onexit cleanup
+
+log_assert "ENOSPC is returned when file system is full."
+
+default_setup_noexit $DISK_SMALL
+log_must zfs set compression=off $TESTPOOL/$TESTFS
+log_must zfs snapshot $TESTPOOL/$TESTFS@snap
+
+log_note "Writing file: $TESTFILE0 until ENOSPC."
+file_write -o create -f $TESTDIR/$TESTFILE0 -b $BLOCKSZ \
+    -c $NUM_WRITES -d $DATA
+ret=$?
+
+(( $ret != $ENOSPC )) && \
+    log_fail "$TESTFILE0 returned: $ret rather than ENOSPC."
+
+log_must zfs umount $TESTPOOL/$TESTFS
+log_must eval "df -h |grep $TESTPOOL"
+size=$(df -h /$TESTPOOL | grep $TESTPOOL | awk '{print $2}')
+used=$(df -h /$TESTPOOL | grep $TESTPOOL | awk '{print $3}')
+if [[ $size = "0" ]] || [[ $used = "0" ]]
+then
+	log_fail "df failed with zero filesystem bytes used"
```

Comment:
Let's log `$size` and `$bytes` in the failure message.

Reply:
Good idea.  Here's what the failure looks like (zfs without the fix).
```
Test: /root/statfs/zfs/tests/zfs-tests/tests/functional/no_space/setup (run as root) [00:00] [PASS]
11:57:30.16 SUCCESS: truncate -s 100m /var/tmp/vdev_small
11:57:30.17 SUCCESS: truncate -s 512m /var/tmp/vdev_large
Test: /root/statfs/zfs/tests/zfs-tests/tests/functional/no_space/enospc_df (run as root) [00:17] [FAIL]
11:57:30.20 ASSERTION: Correct df output is returned when file system is full.
11:57:30.20 NOTE: begin default_setup_noexit
11:57:30.39 SUCCESS: zpool create -f testpool /var/tmp/vdev_small
11:57:30.44 SUCCESS: zfs create testpool/testfs
11:57:30.49 SUCCESS: zfs set mountpoint=/var/tmp/testdir testpool/testfs
11:57:30.52 SUCCESS: zfs set compression=off testpool/testfs
11:57:30.52 NOTE: Writing file: testfile0 until ENOSPC.
11:57:46.53 write failed (-1), good_writes = 4972, error: No space left on device[28]
11:57:46.56 SUCCESS: zfs umount testpool/testfs
11:57:46.56 testpool        896K     0  896K   0% /testpool
11:57:46.57 SUCCESS: eval df -h | grep testpool
11:57:46.57 df failed with size 896K and used 0.
```